packages/@react-spectrum/table/src/Table.tsx

```diff
@@ -80,7 +88,16 @@ function Table<T extends object>(props: SpectrumTableProps<T>, ref: DOMRef<HTMLD
       : DEFAULT_HEADER_HEIGHT[scale],
     estimatedHeadingHeight: props.overflowMode === 'wrap'
       ? DEFAULT_HEADER_HEIGHT[scale]
-      : null
+      : null,
+    getDefaultWidth: ({defaultWidth, width, hideHeader, isSelectionCell}) => {
+      if (!width && hideHeader) {
```

Comment:
why the check for `!width`? i don't know if a column can be zero width, but it's safer to check isNaN rather than !number if it's really needed